### PR TITLE
Magento 2 cron needs zombies to exist

### PR DESCRIPTION
**Pull request by kierenevans**
1. Main cron process spawns new standalone processes as children
2. Main cron process exits after doing it's tasks
3. Zombies created out of the forked processes
4. PID 1 (tini) becomes parent of zombies
5. PID 1 (tini) reaps the zombies
6. Standalone processes never complete

### CODE REVIEW

**Review by tkotosz — DISMISSED** [review]

**Comment by andytson-inviqa** [comment]
if a bash shell for the cron task could run the task then `wait` (waits for all children to finish) then maybe no need

**Comment by kierenevans** [comment]
I think that's what we did in https://github.com/continuouspipe/dockerfiles/pull/180/files with supervisor PID 1, will adjust.

**Comment by andytson-inviqa** [comment]
I think that's just for allowing shell and not just a single command

more like (maybe) `bash -c '... &  && wait'`. I don't know if the cron.php's forks become the bash process's children tho

**Comment by kierenevans** [comment]
Unfortunately they don't: https://github.com/magento/magento2/blob/2.4.0/app/code/Magento/Cron/Observer/ProcessCronQueueObserver.php#L255-L261

**Comment by kierenevans** [comment]
We'll try this where we've been seeing issues :crossed_fingers:

**Comment by andytson-inviqa** [comment]
that's spawned children rather than forks btw

File: src/magento2/harness.yml [hunk 1]
```diff
@@ -153,7 +153,7 @@ attributes:
         - run magento cache:clean
     cron:
       jobs:
-        - "= (@('app.mode') == 'production' ? '* * * * * /cron-run-with-env.sh /app/bin/magento cron:run' : '')"
+        - "= (@('app.mode') == 'production' ? '* * * * * /cron-run-with-env.sh /bin/bash -c \"/app/bin/magento cron:run & && wait\"' : '')"
```
Comment (kierenevans) [inline src/magento2/harness.yml]:
```
bash: -c: line 0: syntax error near unexpected token `&&'
bash: -c: line 0: `/bin/bash -c /app/bin/magento cron:run & && wait'
```

Comment (kierenevans) [reply src/magento2/harness.yml]:
```suggestion
        - "= (@('app.mode') == 'production' ? '* * * * * /cron-run-with-env.sh /bin/bash -m -c \\'\"/app/bin/magento cron:run & wait\"\\'' : '')"
```

**Review by tkotosz — APPROVED** [review]

**Pull request merged**